client/network/src/discovery.rs

```diff
@@ -105,16 +119,9 @@ impl DiscoveryBehaviour {
 			warn!(target: "sub-libp2p", "mDNS is not available on this platform");
 		}
 
-		let local_id = local_public_key.clone().into_peer_id();
-		let store = MemoryStore::new(local_id.clone());
-		let mut kademlia = Kademlia::new(local_id.clone(), store);
-		for (peer_id, addr) in &user_defined {
-			kademlia.add_address(peer_id, addr.clone());
-		}
-
 		DiscoveryBehaviour {
 			user_defined,
-			kademlia,
+			kademlias: HashMap::new(),
```

Comment:
I would always add the default Kademlia here, so that users don't have to worry about that.

Reply:
Is the "default" not a temporary hack that should be removed?

Reply:
Yes, but "temporary" probably means a couple months, which is long enough to do it a bit properly.

Reply:
What do you mean by "which is long enough to do it a bit properly"?